include/z64scene.h

```diff
@@ -1,11 +1,18 @@
 #ifndef Z64SCENE_H
 #define Z64SCENE_H
 
-#include "z64.h"
-#include "z64dma.h" // for RomFile
+#include "macros.h"
+#include "ultra64.h"
+#include "z64bgcheck.h"
+#include "z64dma.h"
```

Comment:
```suggestion
#include "z64dma.h" // for RomFile
```
please don't remove that
ideally please add similar comments to the other includes (except ultra64.h gets a pass), it makes it much easier to maintain includes, but not forcing you

Reply:
This comment was inaccurate, it was used for more than RomFile. There are automated tools for this (e.g. include-what-you-use) and I was going to look into setting them up actually (though no promises). So I don't really want to maintain this manually

Reply:
I had no idea there was a tool for this, didn't even think to ever check.
If it can work then sure, but if not the point stands for maintaining these comments

some notes on include-what-you-use:
- ez install on ubuntu: `apt install iwyu` (uses clang-17 which it will install if not already installed)
- on a single file: `iwyu -D_LANGUAGE_C -Iinclude -Iinclude/libc -fno-builtin -nostdinc include/z64play.h`
- [once errors are fixed, i.e. for z64play.h make z64interface.h include z64view.h]: `-Xiwyu --update_comments` will add the comments I mentioned (for debugging at this point)

however I did not find an option to automatically edit files, idk if that can be done

Reply:
Yeah it's pretty cool. I'm pretty sure it can edit files too (apparently with fix_includes.py). It seems like it requires a fair amount of configuration/setup to get it to do what you want though.

> once errors are fixed, i.e. for z64play.h make z64interface.h include z64view.h
Maybe we could deal with these errors by adding global.h to the top of every file and then running IWYU? Hopefully it'll remove the global.h and replace it with finer-grained headers.

I really don't feel like manually commenting where every single symbol in this file comes from, sorry. I don't like the comments anyway tbh, I think they're noisy and I think it would be better to do an IWYU pass every so often to clean up includes instead of using the comments to maintain includes manually.

Reply:
alright